test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_copy.cpp

```diff
@@ -0,0 +1,502 @@
+//
+// Copyright (c) 2024 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "basic_command_buffer.h"
+#include "procs.h"
+#include <vector>
+
+//--------------------------------------------------------------------------
+namespace {
+
+// CL_INVALID_COMMAND_QUEUE if command_queue is not NULL.
+struct CommandBufferCopyQueueNotNull : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error =
+            clCommandCopyBufferKHR(command_buffer, queue, in_mem, out_mem, 0, 0,
+                                   data_size(), 0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_COMMAND_QUEUE,
+                               "clCommandCopyBufferKHR should return "
+                               "CL_INVALID_COMMAND_QUEUE",
+                               TEST_FAIL);
+
+        const size_t origin[3] = { 0, 0, 0 }, region[3] = { 512, 512, 1 };
+
+        error = clCommandCopyBufferRectKHR(
+            command_buffer, queue, in_mem, out_mem, origin, origin, region, 0,
+            0, 0, 0, 0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_COMMAND_QUEUE,
+                               "clCommandCopyBufferRectKHR should return "
+                               "CL_INVALID_COMMAND_QUEUE",
+                               TEST_FAIL);
+
+        const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+        clMemWrapper image = create_image_2d(
+            context, CL_MEM_READ_WRITE, &formats, 512, 512, 0, NULL, &error);
```

Comment:
We need to check for image support in these tests before calling image creation APIs, probably a similar solution to https://github.com/KhronosGroup/OpenCL-CTS/pull/1944